pkg/metrics/collector.go

```diff
@@ -0,0 +1,107 @@
+package metrics
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	k8smetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/aquasecurity/trivy-operator/pkg/apis/aquasecurity/v1alpha1"
+)
+
+var (
+	imageVulnLabels = []string{
+		"workload_namespace",
+		"workload_name",
+		"image_registry",
+		"image_repository",
+		"image_tag",
+		"image_digest",
+		"severity",
+	}
+	imageVulnDesc = prometheus.NewDesc(
+		prometheus.BuildFQName("trivy", "vulnerabilityreport", "image_vulnerabilities"),
+		"Number of container image vulnerabilities",
+		imageVulnLabels,
+		nil,
+	)
+	severities = map[string]func(vs v1alpha1.VulnerabilitySummary) int{
+		"Critical": func(vs v1alpha1.VulnerabilitySummary) int {
+			return vs.CriticalCount
+		},
+		"High": func(vs v1alpha1.VulnerabilitySummary) int {
+			return vs.HighCount
+		},
+		"Medium": func(vs v1alpha1.VulnerabilitySummary) int {
+			return vs.MediumCount
+		},
+		"Low": func(vs v1alpha1.VulnerabilitySummary) int {
+			return vs.LowCount
+		},
+		"Unknown": func(vs v1alpha1.VulnerabilitySummary) int {
+			return vs.UnknownCount
+		},
+	}
+)
+
+type ResourcesMetricsCollector struct {
+	client.Client
+	Logger logr.Logger
+}
+
+func (c *ResourcesMetricsCollector) SetupWithManager(mgr ctrl.Manager) error {
+	return mgr.Add(c)
+}
+
+func (c ResourcesMetricsCollector) Collect(metrics chan<- prometheus.Metric) {
+	ctx := context.Background()
+	vrList := &v1alpha1.VulnerabilityReportList{}
+	if err := c.List(ctx, vrList, client.InNamespace("")); err != nil {
```

Comment:
seems to me like this `List` call is dependent on the operator's configuration. While most folks will install it globally and this would apply for such cases. There are other configuration modes that might need to be taken into account. Perhaps the `ResourceMetricsCollector` initialization should also take a [Config](https://github.com/aquasecurity/trivy-operator/blob/main/pkg/operator/etc/config.go#L12) object? that way, you'd be able to resolve the installation mode and determine the Namespaces to list VulnerabilityReporLists from.

For deployment modes such as the ones the Operator Lifecycle Manager uses, this is relevant, as the OLM itself determines RBAC permissions based on the target namespaces. And so, for other installation modes, the `List` call might fail.

Reply:
Good catch! I forgot about the other install modes. I will fix this!

Reply:
Target namespaces are now addressed in the PR. PTAL!

Reply:
The latest fix should assure this works with any Client, but I think it is not required when using the catching manager client, which I think should be used anyway....